core/neomutt.c

```diff
@@ -209,3 +210,22 @@ size_t neomutt_mailboxlist_get_all(struct MailboxList *head, struct NeoMutt *n,
 
   return count;
 }
+
+/**
+ * mutt_file_fopen_masked_full - A wrapper around mutt_file_fopen_full with the
+ * mask parameter set to the user's default mode_t permissions, inferred from
+ * the umask
+ */
+FILE *mutt_file_fopen_masked_full(const char *path, const char *mode, const char *file,
+                                  int line, const char *func)
+{
+  // We need to change the current umask to the original one we retrieved from
+  // main.c, _and_ also to use the original umask to calculate the permissions
+  // of the file we create.
+  umask(NeoMutt->user_default_umask);
+  // This 0666 & ~default_umask argument implements the same behavior every unix
+  // utility operates with respect to the shell's umask
+  return mutt_file_fopen_full(path, mode, 0666 & ~NeoMutt->user_default_umask, file, line, func);
+  // Change it quickly back to our default, safe umask of 0700
+  umask(0700);
```

Comment:
If we **do** need them (and I'm missing the point),
then it's important for this `umask()` **NOT** to be after the `return` statement! :-)